fix(install/source) update source installation instructions to use kong build tools (#2144)
Kong/docs.konghq.com: app/install/source.md

{% capture luajit_version %}{{site.data.kong_latest.dependencies.luajit}}{% endcapture %}
{% capture luarocks_version %}{{site.data.kong_latest.dependencies.luarocks}}{% endcapture %}
{% capture openresty_version %}{{site.data.kong_latest.dependencies.openresty}}{% endcapture %}
{% capture openssl_version %}{{site.data.kong_latest.dependencies.openssl}}{% endcapture %}
{% capture libyaml_version %}{{site.data.kong_latest.dependencies.libyaml}}{% endcapture %}
{% capture pcre_version %}{{site.data.kong_latest.dependencies.pcre}}{% endcapture %}

> {% capture gopluginserver_version %}{{site.data.kong_latest.dependencies.gopluginserver}}{% endcapture %}

Reviewer: does this also need to be removed given we are not including go-pluginserver as a dep?

configuration properties at start-up, and the database as storage of all configured entities,
such as the Routes and Services to which Kong proxies.

When not using a database, you will use `environment variables`, and a `kong.yml`
file for specifying the entities as a declarative configuration.

## With a Database

**About the dependencies**

Kong depends on several components (the versions mentioned are the preferred ones):
- [OpenResty {{openresty_version}}](https://openresty.org/en/installation.html).
- [Nginx Patches](https://github.com/Kong/kong-build-tools/tree/master/openresty-build-tools/patches)
- [OpenResty Patches](https://github.com/Kong/kong-build-tools/tree/master/openresty-patches/patches/{{openresty_version}})
- [Kong Nginx Module](https://github.com/Kong/lua-kong-nginx-module/)
- [OpenSSL {{openssl_version}}](https://www.openssl.org/)
- [PCRE {{pcre_version}}](http://www.pcre.org/)
- [LuaRocks {{luarocks_version}}](https://github.com/luarocks/luarocks/wiki/Download)
- [LibYAML {{libyaml_version}}](https://pyyaml.org/wiki/LibYAML)
- [go-pluginserver {{gopluginserver_version}}](https://github.com/Kong/go-pluginserver)
    

> It is not trivial to get everything built correctly, thus we provide a

Reviewer: ```suggestion
It is not trivial to get everything built correctly, thus we provide [kong-build-tools](https://github.com/Kong/kong-build-tools/).
```

[kong-build-tools](https://github.com/Kong/kong-build-tools/).
    
1. **Build the dependencies** 

>    At first, you need to clone the `kong-build-tools`:

Reviewer: ```suggestion
   At first, you need to clone `kong-build-tools`:
```

       --prefix deps \
       --work work \
       --openresty {{openresty_version}} \
       --openssl {{openssl_version}} \
       --kong-nginx-module master \
       --luarocks {{luarocks_version}} \
       --pcre {{pcre_version}} \
       --jobs 6 \
       --force
   ```
   
   Now, you should have the dependencies built, but before we continue,
   let's export some environment variables:
   
   ```bash
   $ export OPENSSL_DIR=$(pwd)/deps/openssl
   $ export PATH=$(pwd)/deps/openresty/bin:$PATH
   $ export PATH=$(pwd)/deps/openresty/nginx/sbin:$PATH
   $ export PATH=$(pwd)/deps/openssl/bin:$PATH
   $ export PATH=$(pwd)/deps/luarocks/bin:$PATH
   ```
   
   Check everything is working:
   
   ```bash
   $ nginx -V
   $ resty -v
   $ openresty -V
   $ openssl version -a
   $ luarocks --version
   ```
   
   You will also need [`LibYAML {{libyaml_version}}`](https://pyyaml.org/wiki/LibYAML)
   that you can install either manually or by using your operating system packages:
   - Ubuntu: `apt install libyaml-dev`
   - Fedora: `dnf install libyaml-devel`
   - macOS: `brew install libyaml`
   
   At this point you may continue to the next step, but if you are interested

>    in running Go plugins, please follow.

Reviewer: Suggestion: let's link to the Go guide instead of writing explicit instructions over here, as the guide contains more info. Something like "At this point you may continue to the next step, but if you are interested in running Go plugins, please follow the [Go guide][go]."

Then we need to add the link at the end
```
[go]: /{{page.kong_version}}/go
```

    example we decided to specify it for illustrative purposes.

6. **Use Kong**

    Verify that Kong is running:

    ```bash
    $ curl --include http://localhost:8001/
    ```

    Quickly learn how to use Kong with the [5-minute Quickstart](/latest/getting-started/quickstart).

## Without a database

1. Follow steps 1 and 2 (Build the dependencies, Install Kong) from the list above.

2. **Write a declarative configuration file**

    The following command will generate a `kong.yml` file in your current folder.
    It contains instructions about how to fill it up. Follow the
    [Declarative Configuration Format]: /{{site.data.kong_latest.release}}/db-less-and-declarative-config/#the-declarative-configuration-format
    instructions while doing so.

    ``` bash
    $ kong config init
    ```

3. **Start Kong**

    On database configuration steps above, we used `kong.conf` file for the configuration.
    Alternatively, you can use environment variables to configure Kong: 

    ```bash
    $ KONG_DATABASE=off \
      KONG_DECLARATIVE_CONFIGURATE=kong.yml \
      kong start --prefix node
    ```

4. **Use Kong**

>     Verify that the Kong is running, and it has the entities detailed in the declarative config file:

Reviewer: ```suggestion
    Verify that Kong is running, and it has the entities detailed in the declarative config file:
```
